### PR TITLE
feat: settings screen initial implementation

### LINKED ISSUE #1163: Capri settings menu
Do the client side work to allow the user to access the menu and perform various experiences critical for Capri launch

## How to find this screen
1. Open LandPKS
2. Select the "Settings" button in the footer

## Design
[Figma](https://www.figma.com/file/KMJbhjVyrrLFHj6jGcX9lN/LandPKS-Design-Prototype-working-file?type=design&node-id=5800-13101&mode=design&t=3XXCUsJRxLmrQY78-0)
#### Design as of 4/22/2024, please view Figma for most up-to-date version
Image

## Acceptance Criteria
- [x] As a logged-in user, I want to be able to access a settings screen where I can
   - [x] see the account I am logged in with
   - [x] See button log out of my own account (tying it to API will be part of https://github.com/techmatters/terraso-product/issues/418)

### PR DESCRIPTION
**Pull request by tm-ruxandra**
## Description

Add the basic settings screen per design requirements. Introduce a new button type (ListButton) and container (ButtonList) to implement its controls, and add support for displaying application version. Refactor the existing logout system to use a common modal, now compliant with design spec, built on the existing ActionModal.

Note that the 'delete account' button is not currently hooked into functionality, as the 'delete' workflow is not yet implemented.

### Checklist
- [x] Corresponding issue has been opened
- [x] English strings reviewed and copyedited
- [x] Strings are localized

### Related Issues
Fixes #1163

### Verification steps

Open settings screen in bottom navbar. Verify expected functionality.

### CODE REVIEW

File: dev-client/src/components/modals/ActionsModal.tsx [hunk 1]
```diff
@@ -113,10 +115,15 @@ export const ActionsModal = forwardRef<
       backgroundColor="grey.200"
       Footer={Footer}
       {...modalProps}>
-      <Heading variant="h5" textAlign="center">
```
Comment (tm-ruxandra) [inline dev-client/src/components/modals/ActionsModal.tsx:116]: The design calls for no header text on the logout confirm modal, so I refactored this system to make it optional.

File: dev-client/src/screens/SettingsScreen/components/UserIndicatorComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import {Text} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {useSelector} from 'terraso-mobile-client/store';
+
+export function UserIndicator() {
+  const currentUser = useSelector(state => state.account.currentUser.data);
+  const {t} = useTranslation();
+
+  return (
+    <Text variant="body1">
+      {currentUser?.email ?? t('settings.unknownUser')}
```
Comment (tm-ruxandra) [inline dev-client/src/screens/SettingsScreen/components/UserIndicatorComponent.tsx:28]: This shouldn't ever happen in practice, but the `currentUser` selector can possibly return an empty value so this seemed good for completion's sake.

File: dev-client/src/screens/SettingsScreen/components/VersionIndicatorComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import {APP_CONFIG} from 'terraso-mobile-client/config';
+
+export function VersionIndicator() {
+  const {t} = useTranslation();
+
+  return (
+    <Text variant="body2">
+      {APP_CONFIG.version
+        ? t('settings.version', {version: APP_CONFIG.version})
+        : `(${t('settings.unknownVersion')})`}
```
Comment (tm-ruxandra) [inline dev-client/src/screens/SettingsScreen/components/VersionIndicatorComponent.tsx:29]: If the app is running locally, it has no tagged version available.

File: dev-client/src/components/modals/ActionsModal.tsx [hunk 4]
```diff
@@ -113,10 +115,15 @@ export const ActionsModal = forwardRef<
...
-        {title}
-      </Heading>
-      <Box height="md" />
+      {hasTitle && (
```
Comment (paulschreiber) [inline dev-client/src/components/modals/ActionsModal.tsx:118]: question: do you need a hasTItle prop? Would it make sense just to check for title being truthy?

File: dev-client/src/translations/en.json [hunk 5]
```diff
@@ -274,6 +274,13 @@
       "nameDesc": "Sort (Z to A)"
     }
   },
+  "settings": {
+    "unknownUser": "Unknown user",
+    "logOut": "Log Out",
+    "deleteAccount": "Delete Account",
+    "unknownVersion": "Untagged application version",
+    "version": "Version {{version}}"
```
Comment (paulschreiber) [inline dev-client/src/translations/en.json:282]:
```suggestion
    "unknownUser": "Unknown user",
    "log_out": "Log Out",
    "delete_account": "Delete Account",
    "unknown_version": "Untagged application version",
    "version": "Version {{version}}"
```

File: dev-client/src/screens/SettingsScreen/components/UserIndicatorComponent.tsx (hunk 2 above)
Comment (paulschreiber) [inline dev-client/src/screens/SettingsScreen/components/UserIndicatorComponent.tsx:28]:
```suggestion
      {currentUser?.email ?? t('settings.unknown_user')}
```

Comment (paulschreiber) [inline dev-client/src/screens/SettingsScreen/components/UserIndicatorComponent.tsx:28]:
```suggestion
      {currentUser?.email ?? t('settings.unknown_user')}
```

File: dev-client/src/screens/SettingsScreen/components/VersionIndicatorComponent.tsx (hunk 3 above)
Comment (paulschreiber) [inline dev-client/src/screens/SettingsScreen/components/VersionIndicatorComponent.tsx:29]:
```suggestion
        : `(${t('settings.unknown_version')})`}
```

File: dev-client/src/screens/SettingsScreen/components/actions/DeleteAccountButton.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import {ListButton} from 'terraso-mobile-client/components/buttons/ListButton';
+
+export function DeleteAccountButton() {
+  const {t} = useTranslation();
+
+  return (
+    <ListButton
+      type="error"
+      iconName="delete"
+      labelText={t('settings.deleteAccount')}
```
Comment (paulschreiber) [inline dev-client/src/screens/SettingsScreen/components/actions/DeleteAccountButton.tsx:28]:
```suggestion
      labelText={t('settings.delete_account')}
```

File: dev-client/src/components/modals/ActionsModal.tsx (hunk 4 above)
Comment (tm-ruxandra) [reply dev-client/src/components/modals/ActionsModal.tsx:118]: It's checked in two places in the jsx and this seemed like it was cleaner.

File: dev-client/src/screens/SettingsScreen/components/VersionIndicatorComponent.tsx (hunk 3 above)
Comment (tm-ruxandra) [reply dev-client/src/screens/SettingsScreen/components/VersionIndicatorComponent.tsx]: I have created https://github.com/techmatters/terraso-mobile-client/issues/1258 to address this issue at some point in the future. The instances introduced in this PR are now fixed.

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**